src/controls/IRoomControllerV2.ts

```diff
@@ -0,0 +1,380 @@
+        // set
+        // modeslist
+
+        // TODO: Subcontrols not yet implemented:
+        // IRCV2Daytimer
+    }
+
+    // Calculate and update tempTargetMin/Max
+    // Should be triggered when any inputs change
+    // TODO: this is called when any of the input variables change in handlers above.
+    // Would be slightly more efficient (& complex) to only call this update if the
+    // inputs relating to current mode change. Ie. dont call this on absentMinOffset
+    // change if mode is 1.
+    protected async updateTempTargetMinMax(): Promise<void> {
+        const activeMode = this.convertStateToInt(this.getCachedStateValue(this.uuid + '.activeMode'));
+        let tempTargetMin;
+        let tempTargetMax;
+        switch (activeMode) {
+            case 0:
+                // Economy, min/max are target +/- absent offset
+                const absentMinOffset = this.convertStateToFloat(
+                    this.getCachedStateValue(this.uuid + '.absentMinOffset'),
+                );
+                const absentMaxOffset = this.convertStateToFloat(
+                    this.getCachedStateValue(this.uuid + '.absentMaxOffset'),
+                );
+                const comfortTemperature = this.convertStateToFloat(
+                    this.getCachedStateValue(this.uuid + '.comfortTemperature'),
+                );
+
+                tempTargetMin = comfortTemperature - absentMinOffset;
+                tempTargetMax = comfortTemperature + absentMaxOffset;
+                this.adapter.log.debug(
+                    `Mode ${activeMode}: ${absentMinOffset}/${absentMaxOffset}/${comfortTemperature} -> ${tempTargetMin}/${tempTargetMax}`,
+                );
+                break;
+
+            case 1:
+            case 3:
+                // Comfort & manual mode just use direct target
+                tempTargetMin = tempTargetMax = this.convertStateToFloat(
+                    this.getCachedStateValue(this.uuid + '.tempTarget'),
+                );
+                this.adapter.log.debug(`Mode ${activeMode}: ${tempTargetMin} -> ${tempTargetMin}/${tempTargetMax}`);
+                break;
+
+            case 2:
+                // Building protection uses frost & heat protection
+                tempTargetMin = this.convertStateToFloat(
+                    this.getCachedStateValue(this.uuid + '.frostProtectTemperature'),
+                );
+                tempTargetMax = this.convertStateToFloat(
+                    this.getCachedStateValue(this.uuid + '.heatProtectTemperature'),
+                );
+                this.adapter.log.debug(`Mode ${activeMode}: -> ${tempTargetMin}/${tempTargetMax}`);
+                break;
+
+            default:
+                this.adapter.log.error(`Unknown IRoomControllerV2 activeMode: ${activeMode}`);
+                break;
+        }
+
+        // Don't do any update if not calculated (can only happen in error condition)
+        if (tempTargetMin != null) {
```

Comment:
Please initialize the variable above with `null` or check for `undefined` here.
Same below

Reply:
No worries. Changed the check to `undefined`.